wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -221,32 +128,46 @@ public static void main(String... args) throws Exception {
     return null;
   }
 
-  public WireCompiler(String protoPath, List<String> sourceFileNames, List<String> roots,
-      String outputDirectory, String registryClass, boolean emitOptions, List<String> enumOptions,
+  @Deprecated
+  public WireCompiler(String protoPath, String outputDirectory, List<String> sourceFileNames,
+      List<String> roots, String registryClass, boolean emitOptions, List<String> enumOptions,
       Constructor<?> serviceWriterConstructor, List<String> serviceWriterOptions) {
-    this(protoPath, sourceFileNames, roots, outputDirectory, registryClass, emitOptions,
-        enumOptions, serviceWriterConstructor, serviceWriterOptions, new IO.FileIO());
+    this(new CommandLineOptions(protoPath, outputDirectory, sourceFileNames, roots,  registryClass,
+        emitOptions,  new HashSet<String>(enumOptions), serviceWriterConstructor.getName(),
+        serviceWriterOptions));
+  }
+
+  public WireCompiler(CommandLineOptions options) {
+    this(options, new IO.FileIO());
   }
 
-  WireCompiler(String protoPath, List<String> sourceFileNames, List<String> roots,
-      String outputDirectory, String registryClass, boolean emitOptions, List<String> enumOptions,
-      Constructor<?> serviceWriterConstructor, List<String> serviceWriterOptions, IO io) {
+  WireCompiler(CommandLineOptions options, IO io) {
+    this.options = options;
+
+    String protoPath = options.protoPath;
+    if (options.javaOut == null) {
+      System.err.println("Must specify " + CommandLineOptions.JAVA_OUT_FLAG + " flag");
+      System.exit(1);
```

Comment:
Good idea.  I'll refactor this to throw an exception.  I moved it because I was thinking that having validation logic in main() is keeping us from having a nice way to invoke the compiler without passing strings for command line options.  The next refactoring moves logging into a logger method so we don't assume console output.